synapse/module_api/__init__.py

```diff
@@ -439,6 +489,144 @@ async def send_local_online_presence_to(self, users: Iterable[str]) -> None:
                     presence_events
                 )
 
+    def background_call_async(self, f: Callable, *args, **kwargs):
+        """Wraps an async function as a background process and runs it.
+
+        Args:
+            f(function): The function to call.
+            *args: Positional arguments to pass to function.
+            **kwargs: Key arguments to pass to function.
+
+        """
+
+        @wrap_as_background_process(f.__name__)
+        async def background_call(*args, **kwargs):
+            await f(*args, **kwargs)
+
+        if self._hs.config.run_background_tasks:
+            self._clock.call_later(0.0, background_call, *args, **kwargs)
+        else:
+            logger.warning(
+                "Not running looping call %s as the configuration forbids it",
+                f,
+            )
```

Comment:
yeah fair enough, I don't really have any great suggestions here. it's probably best to expose a `looping_call` method on ModuleApi which calls `run_as_background_process` on the function (I think you may as well call `run_as_background_process` directly rather than messing with `wrap_as_background_process`).

It might be nice to make the "description" an optional param rather than hardcoding it to the name of the function.